### ISSUE TITLE
Some definitions are not shown

### ISSUE THREAD

**Issue by brupelo** [sublimehq/Packages#712]
Issue based on [this](https://github.com/sublimehq/Packages/issues). Steps to reproduce the bug:

 - Download these 2 files, [mcve](https://gist.github.com/d50ea81619ba3a2f1055bc3e06815c34) and
 [wingdi.h](https://gist.github.com/4e900e650971b999550c33fdaa6e42dc)
 - Create a new workspace on sublime and add the previous 2 files
 - Open the mcve file and move the cursor around like shown [here](https://dl.dropboxusercontent.com/u/21334722/ShareX/2016/11/2016-11-26_16-41-06.gif)

As you can see, neither [wglMakeCurrent](https://gist.github.com/brupelo/4e900e650971b999550c33fdaa6e42dc#file-wingdi-h-L6080) nor [wglCreateContext](https://gist.github.com/brupelo/4e900e650971b999550c33fdaa6e42dc#file-wingdi-h-L6074) are showing up the existing definitions from `wingdi.h` file.

@wbond comment : `I looked at the .h file, and it appears the preprocessor directives at lines 4754-4761 are the issue`

**Comment by wbond** [comment]
This can be reproduced with:

```C
#if defined(_M_CEE)
#undef GetTextMetrics
__inline
BOOL
GetTextMetrics(
    HDC hdc,
    LPTEXTMETRIC lptm
    )
{
#ifdef UNICODE
    return GetTextMetricsW(
#else
    return GetTextMetricsA(
#endif
        hdc,
        lptm
        );
}
#endif  /* _M_CEE */

#endif

/* new GDI */

typedef struct tagDIBSECTION {
    BITMAP       dsBm;
    BITMAPINFOHEADER    dsBmih;
    DWORD               dsBitfields[3];
    HANDLE              dshSection;
    DWORD               dsOffset;
} DIBSECTION, FAR *LPDIBSECTION, *PDIBSECTION;


WINGDIAPI BOOL WINAPI AngleArc( _In_ HDC hdc, _In_ int x, _In_ int y, _In_ DWORD r, _In_ FLOAT StartAngle, _In_ FLOAT SweepAngle);
```

**Comment by wbond** [comment]
Specifically the issue is that there are multiple opening `GetTextMetricsA(`, but only one closing `);`, which throws off the nesting detection that is used to know when the syntax is at the global scope, and functions can be defined. You can "fix" it by adding an additional `)`, and the error color will be removed from `}`.

The preprocessor contexts are funky since you have to try and handle multiple opening `{`, etc. It seems as though right now we may need to add extra complexity to handle mismatches in `(`/`)` pairs also.

Here are the current definitions: https://github.com/sublimehq/Packages/blob/master/C%2B%2B/C.sublime-syntax#L642-L1104. These are repeated in the C++, Objective-C and Objective-C++ syntaxes also.

**Comment by wbond** [comment]
This was fixed by 69166a011bdb9c2494bc892884a43f0f4aca698f, which will be part of the next dev build
